grr/server/grr_response_server/gui/api_plugins/metadata.py

```diff
@@ -51,14 +60,546 @@ class ApiGetOpenApiDescriptionHandler(api_call_handler_base.ApiCallHandler):
   args_type = None
   result_type = ApiGetOpenApiDescriptionResult
 
+  def __init__(self, router):
+    self.router = router
+    self.open_api_obj = None # The main OpenAPI description object.
+    self.schema_objs = None
+
+    self.proto_primitive_types_names = {
+      protobuf2.TYPE_DOUBLE: "protobuf2.TYPE_DOUBLE",
+      protobuf2.TYPE_FLOAT: "protobuf2.TYPE_FLOAT",
+      protobuf2.TYPE_INT64: "protobuf2.TYPE_INT64",
+      protobuf2.TYPE_UINT64: "protobuf2.TYPE_UINT64",
+      protobuf2.TYPE_INT32: "protobuf2.TYPE_INT32",
+      protobuf2.TYPE_FIXED64: "protobuf2.TYPE_FIXED64",
+      protobuf2.TYPE_FIXED32: "protobuf2.TYPE_FIXED32",
+      protobuf2.TYPE_BOOL: "protobuf2.TYPE_BOOL",
+      protobuf2.TYPE_STRING: "protobuf2.TYPE_STRING",
+      protobuf2.TYPE_BYTES: "protobuf2.TYPE_BYTES",
+      protobuf2.TYPE_UINT32: "protobuf2.TYPE_UINT32",
+      protobuf2.TYPE_SFIXED32: "protobuf2.TYPE_SFIXED32",
+      protobuf2.TYPE_SFIXED64: "protobuf2.TYPE_SFIXED64",
+      protobuf2.TYPE_SINT32: "protobuf2.TYPE_SINT32",
+      protobuf2.TYPE_SINT64: "protobuf2.TYPE_SINT64",
+    }
+    self.primitive_types_names = \
+      list(self.proto_primitive_types_names.values()) \
+      + ["BinaryStream",]
+
+  def _SimplifyPathNode(self, node: str) -> str:
+    if len(node) > 0 and node[0] == '<' and node[-1] == '>':
+      node = node[1:-1]
+      node = node.split(":")[-1]
+      node = f"{{{node}}}"
+
+    return node
+
+  def _SimplifyPath(self, path: str) -> str:
+    """Keep only fixed parts and parameter names from Werkzeug URL patterns.
+
+    The OpenAPI specification requires that parameters are surrounded by { }
+    which are added in _SimplifyPathNode.
+    """
+
+    nodes = path.split("/")
+    simple_nodes = [self._SimplifyPathNode(node) for node in nodes]
+
+    simple_path = '/'.join(simple_nodes)
+
+    return simple_path
+
+  def _GetPathArgsFromPath(self, path: str) -> [str]:
+    """Extract path parameters from a Werkzeug Rule URL."""
+    path_args = []
+
+    nodes = path.split("/")
+    for node in nodes:
+      if len(node) > 0 and node[0] == '<' and node[-1] == '>':
+        simple_node = self._SimplifyPathNode(node)
+        simple_node = simple_node[1:-1]
+        path_args.append(simple_node)
+
+    return path_args
+
+  def _GetTypeName(self, t):
```

Comment:
Prefer not to use one-letter identifiers. Since here, `t` is a shortcut for `type` (I presume), the usual Python convention is to use `cls`.

Reply:
Also, try to always use type annotations. If it is not possible to type annotate something, explicitly mark it as `Any` (but here, `Type` would be more strict).

Reply:
Replaced one-letter identifiers and added type annotations throughout the code. Thank you!